types/wlr_surface.c

```diff
@@ -804,8 +804,10 @@ static void subsurface_handle_place_above(struct wl_client *client,
 		return;
 	}
 
+	assert(sibling->parent == subsurface->parent);
```

Comment:
I don't think these asserts are necessary, because `subsurface_find_sibling` already searches in the parent. Or am I missing something?

Reply:
The equivalent sibling searching code that's in wlroots isn't in its own function and where I had put the `assert` line was the closest place that I could put it in to make both of the handler functions(`subsurface_handle_place_above` and `subsurface_handle_place_below`) be on par with what Weston is doing. Sorry if I didn't make that clear at first.

Reply:
Note: the sibling can also be the surfaces parent. I.e. `sibling->parent == subsurface->parent || sibling == subsurface->parent`